benchmarks/src/main/java/io/grpc/benchmarks/driver/LoadServer.java

```diff
@@ -0,0 +1,248 @@
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.grpc.benchmarks.driver;
+
+import com.sun.management.OperatingSystemMXBean;
+
+import io.grpc.Metadata;
+import io.grpc.MethodDescriptor;
+import io.grpc.Server;
+import io.grpc.ServerBuilder;
+import io.grpc.ServerCall;
+import io.grpc.ServerCallHandler;
+import io.grpc.ServerServiceDefinition;
+import io.grpc.Status;
+import io.grpc.benchmarks.ByteBufOutputMarshaller;
+import io.grpc.benchmarks.Utils;
+import io.grpc.benchmarks.proto.BenchmarkServiceGrpc;
+import io.grpc.benchmarks.proto.Control;
+import io.grpc.benchmarks.proto.Messages;
+import io.grpc.benchmarks.proto.Stats;
+import io.grpc.stub.StreamObserver;
+import io.grpc.testing.TestUtils;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.PooledByteBufAllocator;
+import io.netty.util.concurrent.DefaultThreadFactory;
+
+import java.io.File;
+import java.lang.management.ManagementFactory;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Implements the server-side contract for the load testing scenarios.
+ */
+final class LoadServer {
+
+  /**
+   * Generic version of the streaming ping-pong method call.
+   */
+  static MethodDescriptor<ByteBuf, ByteBuf> GENERIC_STREAMING_PING_PONG_METHOD =
+      MethodDescriptor.create(
+          BenchmarkServiceGrpc.METHOD_STREAMING_CALL.getType(),
+          BenchmarkServiceGrpc.METHOD_STREAMING_CALL.getFullMethodName(),
+          new ByteBufOutputMarshaller(),
+          new ByteBufOutputMarshaller());
+
+  private static final Logger LOG = Logger.getLogger(LoadServer.class.getName());
+
+  private final Server server;
+  private final BenchmarkServiceImpl benchmarkService;
+  private final OperatingSystemMXBean osBean;
+  private volatile boolean shutdown;
+  private final int port;
+  private ByteBuf genericResponse;
+  private long lastStatTime;
+  private long lastMarkCpuTime;
+
+  LoadServer(Control.ServerConfig config) throws Exception {
+    LOG.log(Level.INFO, "Server Config \n" + config.toString());
+    port = config.getPort() ==  0 ? TestUtils.pickUnusedPort() : config.getPort();
+    ServerBuilder<?> serverBuilder = ServerBuilder.forPort(port);
+    int asyncThreads = config.getAsyncServerThreads() == 0
+        ? Runtime.getRuntime().availableProcessors()
+        : config.getAsyncServerThreads();
+    switch (config.getServerType()) {
+      case ASYNC_SERVER: {
+        serverBuilder.executor(Executors.newFixedThreadPool(asyncThreads,
```

Comment:
@louiscryan I am still confused why the async servers use a thread pool and the sync server uses a direct executor?